### PR TITLE
added types casting file

### PR DESCRIPTION
**Pull request by SijoGeorge7012**
image
image
image
image
image
image
etc

### CODE REVIEW

File: 004typecasting.html [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        <link rel="stylesheet" href="plugin/highlight/monokai.css" id="highlight-theme" />
+    </head>
+
+    <body>
+        <div class="reveal">
+            <!-- Any section element inside of this container is displayed as a slide -->
+            <div class="slides">
+                <section>
+                    <h2>C# TYPE CONVERSION</h2>
+                    <p>Type conversion is converting one type of data to another type. It is also known as Type Casting. In C#</p>
```
Comment (VishnuKulakkattu) [inline 004typecasting.html:24]: Correct the sentence Please!

**Review by VishnuKulakkattu — CHANGES_REQUESTED** [review]

File: 004typecasting.html [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                </section>
+                <section>
+                    <h3>DIFFERENT TYPES OF CONVERSION</h3>
+                    <p>* Implicit type conversion</p>
+                    <p>* Explicit type conversion</p>
+                </section>
+                <section>
+                    <h3>Implicit type conversion</h3>
+                    <p> These conversions are performed by C# in a type-safe manner. For example, are conversions from smaller to larger integral types and conversions from derived classes to base classes.</p>
```
Comment (wandercode) [inline 004typecasting.html:33]: change text after 'For example'
     to - "conversions from smaller to larger integral types"

**Review by wandercode — CHANGES_REQUESTED** [review]

File: 004typecasting.html [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                </section>
+                <section data-auto-animate>
+                    <h4 data-id="code-title">Example of Implicit type conversion</h4>
+                    <pre data-id="code-animation"><code class="hljs" data-trim data-line-numbers>
+                        int myInt = 9;
+                        double myDouble = myInt;//Automatic casting int to double
+
+                        Console.WriteLine(myInt);//Output 9
+                        Console.WriteLine(myDouble);//Output 9
```
Comment (wandercode) [inline 004typecasting.html:42]: please add right and wrong syntax for every examples .
please dont forget to mention reason for the error in wrong syntax and also to display the output.

**Review by wandercode — CHANGES_REQUESTED** [review]

File: 004typecasting.html [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+                    <h4 data-id="code-title">Example of Type conversion method</h4>
+                    <pre data-id="code-animation"><code class="hljs" data-trim data-line-numbers>
+                        int myInt = 10;
+                        double myDouble = 5.25;
+                        bool myBool = true;
+
+                        Console.WriteLine(Convert.ToString(myInt));    // convert int to string
+                        Console.WriteLine(Convert.ToDouble(myInt));    // convert int to double
+                        Console.WriteLine(Convert.ToInt32(myDouble));  // convert double to int
+                        Console.WriteLine(Convert.ToString(myBool));   // convert bool to string
```
Comment (wandercode) [inline 004typecasting.html:106]: please show the outputs

File: 004typecasting.html [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+                    <h4 data-id="code-title">Example of Type conversion method</h4>
+                    <pre data-id="code-animation"><code class="hljs" data-trim data-line-numbers>
+                        int myInt = 10;
+                        double myDouble = 5.25;
+                        bool myBool = true;
+
+                        Console.WriteLine(Convert.ToString(myInt))    // wrong declaration to convert int to string, missing ;
+                        Console.WriteLine(Convert.ToDouble(my));    // wrong declaration to convert int to double, unassign variable
+                        Console.WriteLine(ConvertToInt32(myDouble));  // wrong declaration to convert double to int, missing . in between Convert and ToInt32
+                        Console.WriteLine(Convert.ToString(myBoo));   // wrong declaration to convert bool to string, unassign variable
```
Comment (wandercode) [inline 004typecasting.html:120]: please show the output

**Review by wandercode — CHANGES_REQUESTED** [review]

File: 004typecasting.html [hunk 6]
```diff
@@ -121,6 +126,29 @@ <h4 data-id="code-title">Example of Type conversion method</h4>
...
+                    <pre data-id="code-animation"><code class="hljs" data-trim data-line-numbers>
+                        int number = Int32.Parse(“123”);
+                    </code></pre>
+                    <p>
+                        Here we converted a string into an integer by using parse. <br>
+                        Let’s look at another conversion method which is the Convert method. <br>
+                    </p>
+                    <p>
+                        Static methods present inside the Convert class are quite useful for converting to the base data type or vice versa. Some of the supported data types are Char, Boolean, Int32, int64, Double, Decimal, String, Int16, etc. Convert class also supports methods for other conversions.
+                    </p>
```
Comment (wandercode) [inline 004typecasting.html:149]: please add examples for each Int32.parse, Double.parse, float.parse etc, (right and wrong syntax)

**Review by wandercode — CHANGES_REQUESTED** [review]

File: 004typecasting.html [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+                        Console.WriteLine(Convert.ToString(myBoo));   // wrong declaration to convert bool to string, unassign variable 
+                    </code></pre>
+                </section>
+            </section>
+            <section>
+                <section>
+                <h4>Conversion Using Different Helper Classes</h4>
+                <p>
+                    To convert between different non-compatible types such as converting a string to a number or a byte array into an integer or even hexadecimal strings into other numeric types, we need different helper class as a direct conversion is not possible.
+                </p>
```
Comment (wandercode) [inline 004typecasting.html:134]: please reduce the size of  paragraph.

File: 004typecasting.html [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+              </section>
+              <section>
+                <h3>Exercise 2</h3>
+                <p>
+                  <ol>
+                    Q1 : Write a program, double a=100.25 and store to int b 
+                  </ol>
+                </p>
+              </section>
+            </div>
```
Comment (wandercode) [inline 004typecasting.html:293]: please add excercises for .Parse method

**Review by wandercode — CHANGES_REQUESTED** [review]

**Comment by wandercode** [comment]
LGTM

**Review by wandercode — APPROVED** [review]

File: 004typecasting.html [hunk 9]
```diff
@@ -128,11 +128,11 @@ <h4 data-id="code-title">Example of Type conversion method</h4>
             </section>
             <section>
                 <section>
-                <h4>Conversion Using Different Helper Classes</h4>
-                <p>
+                <h3>Conversion Using Different Helper Classes</h3>
+                <p style="font-size:30px;">
```
Comment (VishnuKulakkattu) [inline 004typecasting.html:132]: Please move these styling to css, in every code

**Review by VishnuKulakkattu — CHANGES_REQUESTED** [review]

**Comment by wandercode** [comment]
LGTM

**Comment by aneeshs005** [comment]
@SijoGeorge7012 You should include more slides for when the implicit casting occurs when explicit casting occurs. (Eg shown in w3 school)
Slides for different ways for casting. Eg: casting using (int), (double), Convert function, Parse method, TryParse method
What is the common error that occurs when you don't do explicit conversion correctly?

File: 004typecasting.html [hunk 10]
```diff
@@ -0,0 +1,535 @@
...
+                    <h4 data-id="code-title">Conversion Using Different Helper Classes</h4>
+                    <pre data-id="code-animation"><code class="hljs" data-trim data-line-numbers>
+                        using System;
+                        class Program
+                        {
+                            static void Main(string[] args)
+                            {
+                                string a= 10.56; // missing ""
+	
+                                Console.WriteLine(Double.parse(a)); // Double.Parse not Double.parse
```
Comment (aneeshs005) [inline 004typecasting.html:480]: You have given right syntax as wrong syntax

**Review by aneeshs005 — COMMENTED** [review]
Where are the 6 exercises which you did?
I asked you to add that as well, right?

**Comment by aneeshs005** [comment]
@wandercode @VishnuKulakkattu @rohithkrajan Looks good to me

**Review by rohithkrajan — APPROVED** [review]

**Review by wandercode — APPROVED** [review]

**Review by wandercode — APPROVED** [review]

**Review by VishnuKulakkattu — APPROVED** [review]

**Pull request merged**